datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1515,8 +1515,30 @@ pub struct TableScan {
     pub fetch: Option<usize>,
 }
 
+impl PartialEq for TableScan {
+    fn eq(&self, other: &Self) -> bool {
+        self.table_name == other.table_name
```

Comment:
I think this also needs to check that the `source` is equal as well -- I think we can do so via https://doc.rust-lang.org/std/sync/struct.Arc.html#method.ptr_eq

Reply:
This `Arc::ptr_eq` might be **risky**... [Arc::ptr_eq](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.ptr_eq) and [std::ptr::eq](https://doc.rust-lang.org/std/ptr/fn.eq.html) say that `dyn` trait comparisons are **unreliable**. 😕

Even clippy gives an error [vtable_address_comparisons](https://rust-lang.github.io/rust-clippy/master/index.html#vtable_address_comparisons) from **correctness** 🔞 category.

I suggest to reconsider the request about `source` comparison.

Reply:
Upon consideration within a single plan, the `table_name` should be unique. I was originally worried about the case where two TableScan's that had different `source`s but all other fields are the same resulting in a false positive.